test/conversion_test.go

```diff
@@ -0,0 +1,787 @@
+    - name: name
+      type: string
+      value: pr
+    - name: kind
+      type: string
+      value: Task
+  params:
+  - name: STRING_LENGTH
+    value: 1
+    type: string
+  timeouts:
+    pipeline: 60s
+  workspaces:
+  - name: password-vault
+    secret:
+      secretName: secret-password
+  taskRunTemplate: {
+    serviceAccountName: "default"
+  }
+`
+)
+
+// TestTaskCRDConversion first creates a v1beta1 Task CRD using v1beta1Clients and
+// requests it by v1Clients to compare with v1 if the conversion has been
+// correctly executed by the webhook. And then it creates the v1 Task CRD using v1Clients
+// and requests it by v1beta1Clients to compare with v1beta1.
+func TestTaskCRDConversion(t *testing.T) {
+	ctx := context.Background()
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	t.Parallel()
+
+	c, namespace := setup(ctx, t)
+	knativetest.CleanupOnInterrupt(func() { tearDown(ctx, t, c, namespace) }, t.Logf)
+	defer tearDown(ctx, t, c, namespace)
+
+	v1beta1TaskName := helpers.ObjectNameForTest(t)
+	v1beta1Task := parse.MustParseV1beta1Task(t, fmt.Sprintf(v1beta1TaskYaml, v1beta1TaskName, namespace))
+	v1TaskExpected := parse.MustParseV1Task(t, fmt.Sprintf(v1TaskYaml, v1beta1TaskName, namespace))
+
+	if _, err := c.V1beta1TaskClient.Create(ctx, v1beta1Task, metav1.CreateOptions{}); err != nil {
+		t.Fatalf("Failed to create v1beta1 Task: %s", err)
+	}
+
+	v1TaskGot, err := c.V1TaskClient.Get(ctx, v1beta1TaskName, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Couldn't get expected v1 Task for %s: %s", v1beta1TaskName, err)
+	}
+
+	if d := cmp.Diff(v1TaskExpected, v1TaskGot, filterTypeMeta, filterObjectMeta); d != "" {
+		t.Fatalf("-want, +got: %v", d)
+	}
+
+	v1TaskName := helpers.ObjectNameForTest(t)
+	v1Task := parse.MustParseV1Task(t, fmt.Sprintf(v1TaskYaml, v1TaskName, namespace))
+	v1beta1TaskExpected := parse.MustParseV1beta1Task(t, fmt.Sprintf(v1beta1TaskYaml, v1TaskName, namespace))
+
+	if _, err := c.V1TaskClient.Create(ctx, v1Task, metav1.CreateOptions{}); err != nil {
+		t.Fatalf("Failed to create v1beta1 Task: %s", err)
+	}
+
+	v1beta1TaskGot, err := c.V1beta1TaskClient.Get(ctx, v1TaskName, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Couldn't get expected v1beta1 Task for %s: %s", v1TaskName, err)
+	}
+
+	if d := cmp.Diff(v1beta1TaskExpected, v1beta1TaskGot, filterTypeMeta, filterObjectMeta); d != "" {
+		t.Fatalf("-want, +got: %v", d)
+	}
+}
+
+// TestTaskRunCRDConversion first creates a v1beta1 TaskRun CRD using v1beta1Clients
+// and requests it by v1Clients to compare with v1 if the conversion has been correctly
+// executed by the webhook. And then it creates the v1 TaskRun CRD using v1Clients
+// and requests it by v1beta1Clients to compare with v1beta1.
+func TestTaskRunCRDConversion(t *testing.T) {
+	ctx := context.Background()
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	t.Parallel()
+
+	c, namespace := setup(ctx, t, requireAnyGate(supportedV1ConversionFeatureGates))
```

Comment:
Any reason for including `alpha` feature gate for taskRun and pipelineRun conversion here? Do we have any alpha API in the test case? I suggest keeping alpha APIs as a separate test case if that's not the case.

Reply:
Agreed with the separation for better delegations.
This is another deprecated field `bundle` that needs git resolver bundle here. The `bundle` would requires the `alpha` feature flag here.